libs/core/datastructures/include/hpx/datastructures/detail/small_vector.hpp

```diff
@@ -294,22 +301,10 @@ namespace hpx::detail {
                 reinterpret_cast<T*>(m_data.data() + std::alignment_of_v<T>));
         }
 
-        static void uninitialized_move_and_destroy(
-            T* source_ptr, T* target_ptr, std::size_t size) noexcept
-        {
-            if constexpr (std::is_trivially_copyable_v<T>)
-            {
-                std::memcpy(target_ptr, source_ptr, size * sizeof(T));
-            }
-            else
-            {
-                std::uninitialized_move_n(source_ptr, size, target_ptr);
-                std::destroy_n(source_ptr, size);
-            }
-        }
-
         void realloc(std::size_t new_capacity)
         {
+            static_assert(!emulate_inplace_vector,
+                "If called in an inplace_vector, it is a bug.");
```

Comment:
FWIW, `std::inplace_vector` allows you to call `.reserve`. It's a no-op for small capacities, and invariably throws `bad_alloc` (or `length_error`? I'm not sure even LWG knows yet) for large capacities. I don't know if you consider that an improvement or not.